src/sciline/pipeline.py

```diff
@@ -50,6 +53,66 @@ def _is_multiple_keys(keys: type | Iterable[type] | UnionType) -> bool:
     )
 
 
+def _find_paths_to_targets(
+    graph: Any, missing: Any, targets: tuple[Any, ...]
+) -> list[list[Any]]:
+    """Find paths from missing node to each target.
+
+    Parameters
+    ----------
+    graph:
+        The graph to traverse
+    missing:
+        The missing node
+    targets:
+        The target nodes
+
+    Returns
+    -------
+    :
+        List of paths, where each path is a list of nodes from missing to target
+    """
+
+    paths = []
+    for target in targets:
+        try:
+            # Find all simple paths from the missing node to the target
+            target_paths = list(all_simple_paths(graph, missing, target))
+            # Sort by length to show the shortest path first
+            paths.extend(sorted(target_paths, key=len))
```

Comment:
This can still result in a longer path being shown first. If `targets[0]` produces a path of length 4 and `targets[1]` produces a path of length 2, then we get the length 4 path first in the output. So can you sort the final paths instead of each subset?